NitroxServer/Player.cs

```diff
@@ -23,13 +23,13 @@ public class Player : IProcessorContext
         public NitroxId GameObjectId { get; }
         public Optional<NitroxId> SubRootId { get; set; }
 
-        public Player(PlayerContext playerContext, NitroxConnection connection, Vector3 position, Optional<NitroxId> subRootId)
+        public Player(PlayerContext playerContext, NitroxConnection connection, Vector3 position, Optional<NitroxId> subRootId, NitroxId gameObjectId)
         {
             PlayerContext = playerContext;
             this.connection = connection;
             Position = position;
             SubRootId = subRootId;
-            GameObjectId = new NitroxId();
+            GameObjectId = gameObjectId;
```

Comment:
Do we need to have the overhead of persisting this data?  Can we make it deterministic.... maybe based on the player name?  I believe we used to do that but it fell off at some point in time.

Reply:
I've changed the code to use a deterministic NitroxId following the RFC guidelines for GUID version 5.